tiled/access_control/access_policies.py

```diff
@@ -412,3 +414,167 @@ async def filters(
 
         queries.append(query_filter(identifier, tag_list))
         return queries
+
+
+class Data(BaseModel):
+    token: str
+    audience: Optional[str]
+    access_blob: Optional[AccessBlob]
+
+
+class Input(BaseModel):
+    input: Data
+
+
+class Decision(BaseModel):
+    result: Union[List[str], bool]
+
+
+class ExternalPolicyDecisionPoint(AccessPolicy):
```

Comment:
@nmaytan and I wonder if the `TagBasedAccessPolicy` could be reused, using the `tag_parser` argument to inject OPA-specific integration (`_get_external_decision`). Most of the `AccessPolicy` interface will be the same across our local solution, OPA, OpenFGA, and others. The `TagParser` abstraction might be a more tightly-scoped way of injecting framework-specific integration.

Nate will aim to find the bandwidth to implement this suggestion as a PR into your PR. But I mention this now in case you have any immediate thoughts on this.